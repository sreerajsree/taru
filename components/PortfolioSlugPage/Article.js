import DateFormatter from '../Generic/DateFormatter';
import PrintMarkdownHTML from '../Generic/PrintMarkdownHTML';
import Link from '../Generic/NoScrollLink';

export default function Article({
  title, imgSrc, imgSrcSet, date, content,
}) {
  return (
    <article className="flex flex-col items-center justify-center text-center pt-6 md:pt-32">
      <img
        src={imgSrc}
        srcSet={imgSrcSet}
        alt={title}
        className="md:h-50vh"
      />
      <h1 className="text-4xl md:text-6xl font-title text-pink-custom mt-4 md:mt-16 mb-4">{title}</h1>
      <p className="font-title my-2 md:my-8 text-lg">
        <DateFormatter dateString={date} />
      </p>
      <PrintMarkdownHTML content={content} />
      <Link href={{ pathname: '/portfolio', query: { internalLink: true } }}>
        <a className="px-4 py-2 my-8 md:my-8 border-solid border border-black font-body text-base hover:bg-black hover:text-pink-custom ease-in-out transition duration-500">
          Back to Portfolio
        </a>
      </Link>
    </article>
  );
}
