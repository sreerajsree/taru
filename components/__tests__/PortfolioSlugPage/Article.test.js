import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import * as nextRouter from 'next/router';
import * as reactIntersectionObserver from 'react-intersection-observer';

import Article from '../../PortfolioSlugPage/Article';

jest.mock('react-use');
nextRouter.useRouter = jest.fn();
reactIntersectionObserver.useInView = jest.fn().mockReturnValue([]);

nextRouter.useRouter.mockImplementation(() => ({ route: '/', query: {} }));

const title = 'page title';
const image = 'image.jpg';
const date = '2019-10-10T00:00:00';
const content = '<p>content</p>';

describe('Hero', () => {
  test('renders and matches snapshot', () => {
    const { asFragment } = render(
      <Article title={title} image={image} date={date} content={content} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Check that the title, date and content are correctly rendered', () => {
    render(<Article title={title} image={image} date={date} content={content} />);
    expect(screen.getByText('page title')).toBeInTheDocument();
    expect(screen.getByText('October, 2019')).toBeInTheDocument();
    expect(screen.getByText('content')).toBeInTheDocument();
  });
});
