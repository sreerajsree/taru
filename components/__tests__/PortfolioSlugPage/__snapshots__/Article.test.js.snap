// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders and matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="flex flex-col items-center justify-center text-center pt-6 md:pt-32"
  >
    <img
      alt="page title"
      class="md:h-50vh"
    />
    <h1
      class="text-4xl md:text-6xl font-title text-pink-custom mt-4 md:mt-16 mb-4"
    >
      page title
    </h1>
    <p
      class="font-title my-2 md:my-8 text-lg"
    >
      <time
        datetime="2019-10-10T00:00:00"
      >
        October, 2019
      </time>
    </p>
    <div
      class="markdown"
    >
      <p>
        content
      </p>
    </div>
    <a
      class="px-4 py-2 my-8 md:my-8 border-solid border border-black font-body text-base hover:bg-black hover:text-pink-custom ease-in-out transition duration-500"
      href="/portfolio?internalLink=true"
    >
      Back to Portfolio
    </a>
  </article>
</DocumentFragment>
`;
